vital/types/point.h

```diff
@@ -0,0 +1,114 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the interface to a simple point.
+ */
+
+#ifndef KWIVER_VITAL_POINT_H_
+#define KWIVER_VITAL_POINT_H_
+
+#include <vital/vital_config.h>
+#include <vital/vital_export.h>
+#include <vital/types/vector.h>
+#include <vital/types/covariance.h>
+
+namespace kwiver {
+namespace vital {
+
```

Comment:
Since this is more than a simple point, it would be helpful to document what this is and why you would want top use it. That sort of thing.

Reply:
@linus-sherrill 
Is there a definition of a simple point somewhere?

Reply:
we usually just use `vector_2d` or `vector_3d`.  The point of this class (no pun intended) seems to be to bundle a point with a covariance.

Reply:
Your point that this point is a point with a covariance is an accurate point.
Should I rename?

Reply:
Note that we use locations with covariance elsewhere in KWIVER for TeleSculptor.  The "feature" class contains an image location (vector_2d) and covariance plus other things.  The "landmark" contains a 3d location and covariance plus other things.

That's not to say we shouldn't have an "uncertain point" class like this, but we might want to update those other cases to use this class for consistency.

Reply:
I think Linus' comment re: adding documentation still applies?